src/plugins/data/public/utils/shallow_equal.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+function is(x: unknown, y: unknown) {
+  if (x === y) {
+    return x !== 0 || y !== 0 || 1 / x === 1 / y;
+  } else {
+    return x !== x && y !== y;
+  }
+}
+
+/**
+ * Shallow Equal check
+ *
+ * Adapted from react-redux
+ **/
+export function shallowEqual(objA: unknown, objB: unknown): boolean {
```

Comment:
I needed this logic twice. In some other places in kibana this func is imported from `react-redux`. I didn't import it into data plugin because it pulls in `react-redux` into data's bundle

Reply:
I think we should add a comment explaining that this code was copied from a lib.

If you'd prefer to keep the code as-is from the lib, feel free to ignore Object.is comment.